### PR TITLE
Allow filter elasticsearch request for security reasons

### PR DESCRIPTION
**Pull request by seSze**
Added 3 configuration values:

```{
  "elasticsearch":
    "useRequestFilter": false,
    "overwriteRequestSourceParams": false,
    "requestParamsBlacklist": [], 
  }
}
```

`useRequestFilter` is responsible for enabling/disabling ES request filtering
`overwriteRequestSourceParams` decide if you want to overwrite or merge values
`requestParamsBlacklist` allow you to filter out selected attributes

### CODE REVIEW

File: src/api/catalog.ts [hunk 1]
```diff
@@ -110,6 +111,36 @@ export default ({config, db}) => async function (req, res, body) {
   const s = Date.now()
   const reqHash = sha3_224(`${JSON.stringify(requestBody)}${req.url}`)
   const dynamicRequestHandler = () => {
+    // filter request parameters
```
Comment (gibkigonzo) [inline src/api/catalog.ts:114]: wdyt about moving all of this to separate function
```
function filterRequestParams (url, req, entityType, config) {
  if (config.elasticsearch.useRequestFilter && typeof config.entities[entityType] === 'object') {
  ...
    return `${urlParts[0]}?${querystring.stringify(urlParams)}`
  } else {
    return url
  }
}
```
and then fire it in `adjustBackendProxyUrl` ?
```
function adjustBackendProxyUrl (req, indexName, entityType, config) {
  ...
  url = filterRequestParams(url, entityType, config)
  return url
}
```

Comment (seSze) [reply src/api/catalog.ts:114]: sure, will do. That was a quickfix for one of the clients, so I've just copy/paste the solution

**Comment by seSze** [comment]
@gibkigonzo is that ok now?

**Comment by gibkigonzo** [comment]
Awesome :) Thank you

**Pull request merged**